storagenode/pieces/cache.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package pieces
+
+import (
+	"context"
+	"sync"
+
+	"go.uber.org/zap"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/storage"
+)
+
+// StoreWithCache has access to the pieces store and also keeps track of real time
+// totals of space used for all pieces and for all pieces by satellite
+type StoreWithCache struct {
+	Store
+
+	cacheMutex sync.RWMutex
+	cache      spaceUsedCache
+}
+
+type spaceUsedCache struct {
```

Comment:
You can inline these above since there is a separate struct for representing the idea of a cache.